api/internal/target/kusttarget.go

```diff
@@ -422,11 +439,25 @@ func (kt *KustTarget) accumulateDirectory(
 }
 
 func (kt *KustTarget) accumulateFile(
-	ra *accumulator.ResAccumulator, path string) error {
+	ra *accumulator.ResAccumulator, path string, origin *resource.Origin) error {
 	resources, err := kt.rFactory.FromFile(kt.ldr, path)
 	if err != nil {
 		return errors.Wrapf(err, "accumulating resources from '%s'", path)
 	}
+	if kt.kustomization.BuildOptions != nil && kt.kustomization.BuildOptions.AnnoOrigin {
+		var anno string
+		anno = anno + "path: " + filepath.Join(origin.Path, path) + "\n"
+		if origin.Repo != "" {
+			anno = anno + "repo: " + origin.Repo + "\n"
+		}
+		if origin.Ref != "" {
+			anno = anno + "ref: " + origin.Ref + "\n"
+		}
```

Comment:
Likewise, this anno construction code should live in origin, with regression coverage (examples)  in origin_test.go

Reply:
Question: Do you think that origin belongs api/resource, or should would it make more sense to put it in api/types?